### PR TITLE
Fix type annotation

### PR DESCRIPTION
**Pull request by taniokay**
## Motivation

Resolve https://github.com/optuna/optuna/issues/4508

## Description of the changes

Update type annotations in `optuna/testing` according to the following command:
```bash
echo -n '\n[tool.ruff.isort]\nrequired-imports = ["from __future__ import annotations"]\n' >> pyproject.toml && \
ruff --fix --select UP007,UP006,F401,I002, --target-version py37 optuna/testing/ && \
git restore pyproject.toml && \
isort . && \
black .
```

### CODE REVIEW

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Comment by contramundum53** [comment]
@gen740 Could you review this PR?

File: optuna/testing/storages.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 from typing import Any
 from typing import IO
 from typing import Optional
```
Comment (gen740) [inline optuna/testing/storages.py:7]: I guess this Optional can be removed by using the `SomeType | None` syntax.

File: optuna/testing/storages.py [hunk 2]
```diff
@@ -5,7 +5,6 @@
...
-from typing import Type
 from typing import Union
```
Comment (gen740) [inline optuna/testing/storages.py:8]: ditto.

**Comment by taniokay** [comment]
Thank you both for reviewing! Could you take another look?

**Review by gen740 — APPROVED** [review]
LGTM!

**Comment by gen740** [comment]
@contramundum53 Could you merge this PR?

**Pull request merged**